cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpCredentialConnectorTest.java

```diff
@@ -0,0 +1,266 @@
+package com.sequenceiq.cloudbreak.cloud.gcp;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+import javax.ws.rs.BadRequestException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.api.client.testing.http.MockHttpTransport;
+import com.google.api.client.testing.json.MockJsonFactory;
+import com.google.api.services.compute.Compute;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.gcp.context.GcpContext;
+import com.sequenceiq.cloudbreak.cloud.gcp.context.GcpContextBuilder;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredentialStatus;
+import com.sequenceiq.cloudbreak.cloud.model.CredentialStatus;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GcpCredentialConnectorTest {
+
+    @InjectMocks
+    private GcpCredentialConnector underTest;
+
+    @Mock
+    private GcpContextBuilder contextBuilder;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private GcpContext context;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    /**
```

Comment:
Comments are bad. If we don't understand what it does from the name and/or content of the method, the test is bad.